### PR TITLE
[MatrixCamp] Add built-in funcion date

### PR DESCRIPTION
**Pull request by Charlie17Li**
**What type of PR is this?**

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [x] Feature
- [ ] Test and CI
- [ ] Code Refactoring

**Which issue(s) this PR fixes:**

issue #1974 

**What this PR does / why we need it:**

add the built-in func: date

**Special notes for your reviewer:**

Not Available

**Additional documentation (e.g. design docs, usage docs, etc.):**

Not Available

### CODE REVIEW

**Comment by domingozhang** [comment]
@Charlie17Li 
I ran your branch and found the result was not as expected. Please check. thanks.

In MySQL:
WeChat2379afa6fdff6c88220e7ffe6384ff04

In your branch:
WeChate5949d0804202f99bec02bb0cc32b5ca


commit ef968ee9a3538dbcd51b9d4cec5bd6dd85610481 (HEAD -> date, origin/date)
Author: charlie <qianglin98@qq.com>
Date:   Thu Mar 31 13:26:03 2022 +0800

File: pkg/builtin/unary/date.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+	}
+	overload.OpTypes[builtin.Date] = overload.Unary
+	overload.UnaryOps[builtin.Abs] = []*overload.UnaryOp{
+		{
+			Typ:        types.T_date,
+			ReturnType: types.T_date,
+			Fn: func(lv *vector.Vector, proc *process.Process, _ bool) (*vector.Vector, error) {
+				lvs := lv.Col.([]types.Date)
+				size := types.T(types.T_date).TypeLen()
+				vec, err := process.Get(proc, int64(size)*int64(len(lvs)), types.Type{Oid: types.T_date, Size: int32(size)})
```
Comment (broccoliSpicy) [inline pkg/builtin/unary/date.go:34]: perhaps we could reuse the original vector when possible

File: pkg/builtin/unary/date.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
+package unary
```
Comment (broccoliSpicy) [inline pkg/builtin/unary/date.go:1]: could you please add the license header...

**Review by broccoliSpicy — CHANGES_REQUESTED** [review]

File: pkg/builtin/unary/date.go (hunk 1 above)
Comment (Charlie17Li) [reply pkg/builtin/unary/date.go:53]: I have added the code like in abs function:
```
if lv.Ref == 1 || lv.Ref == 0 {
    lv.Ref = 0
    date.DateToDate(lvs, lvs)
    return lv, nil
}
```

but I don't know what's Ref mean (reference count)

File: pkg/builtin/unary/date.go (hunk 2 above)
Comment (Charlie17Li) [reply pkg/builtin/unary/date.go:15]: Done

File: pkg/builtin/unary/date.go (hunk 1 above)
Comment (broccoliSpicy) [reply pkg/builtin/unary/date.go:53]: [PR#2037](https://github.com/matrixorigin/matrixone/pull/2037)
the discussion in this PR may offer some reference for this Ref field

**Review by broccoliSpicy — COMMENTED** [review]
looks good to me

**Pull request merged**